### PR TITLE
Introduce external projects

### PR DESCRIPTION
**Pull request by kleingeist**
External projects inherit from regular projects and thus contain a
blueprint, a module and a phase. To make the handling and styling more
robust the external projects duration is stored in its phase as start
and end date.

External projects are using a different dashboard form, which can be
accesed from the regular select blueprint view.

For convenience a `extproject_url` filter and a `is_external` tag have
been introduced for use in templates.

### CODE REVIEW

File: meinberlin/static/images/external_project.svg [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="50px" height="50px" viewBox="0 0 50 50" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <!-- Generator: Sketch 43.1 (39012) - http://www.bohemiancoding.com/sketch -->
+    <title>projectOwner / external</title>
+    <desc>Created with Sketch.</desc>
+    <defs></defs>
+    <g id="Symbols" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <g id="projectOwner-/-external">
+            <path d="M50,25 C50,38.8070912 38.8072642,50 25.0001114,50 C11.1929587,50 0,38.8070912 0,25 C0,11.1929088 11.1929587,0 25.0001114,0 C38.8072642,0 50,11.1929088 50,25" id="Fill-1" fill="#F3F3F3"></path>
+            <text id="share---fontawesome" font-family="FontAwesome" font-size="24" font-weight="normal" fill="#CD0055">
```
Comment (vellip) [inline meinberlin/static/images/external_project.svg]: @damusp That Icon is not exported as paths.

**Comment by kleingeist** [comment]
has to be adapted to #214 to maintain the active menu item

File: apps/dashboard/forms.py [hunk 2]
```diff
@@ -324,3 +326,71 @@ def save(self, commit=True):
...
+        model = extproject_models.ExternalProject
+        fields = ['name', 'url', 'description', 'image', 'is_archived'] \
+            + ['start_date', 'end_date']
+
+
+class ExternalProjectCreateForm(ExternalProjectBaseForm):
+
+    def __init__(self, view, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.view = view
```
Comment (2e2a) [inline apps/dashboard/forms.py]: As discussed, its probably better to pass all variables required by the form more explicitly via get_form_kwargs.

**Comment by kleingeist** [comment]
@2e2a i modified the form to use individal kwargs

File: apps/dashboard/views.py [hunk 3]
```diff
@@ -47,6 +47,23 @@ class DashboardBlueprintListView(dashboard_mixins.DashboardBaseMixin,
     permission_required = 'meinberlin_organisations.initiate_project'
 
 
+class DashboardProjectCreateViewDispatcher(generic.View):
+
+    def __init__(self, **kwargs):
+        super(DashboardProjectCreateViewDispatcher, self).__init__(**kwargs)
+        self._initkwargs = kwargs
```
Comment (2e2a) [inline apps/dashboard/views.py]: As dsicussed, we can remove the kwargs handling as long as we don't use them.

File: apps/dashboard/urls.py [hunk 4]
```diff
@@ -23,11 +23,15 @@
...
-        views.DashboardProjectCreateView.as_view(),
+        views.DashboardProjectCreateViewDispatcher.as_view(),
         name='dashboard-project-create'),
     url(r'^(?P<organisation_slug>[-\w_]+)/projects/(?P<slug>[-\w_]+)/$',
         views.DashboardProjectUpdateView.as_view(),
         name='dashboard-project-edit'),
+    url(r'^(?P<organisation_slug>[-\w_]+)/'
+        r'external_projects/(?P<slug>[-\w_]+)/$',
+        views.DashboardExternalProjectUpdateView.as_view(),
+        name='dashboard-external-project-edit'),
```
Comment (2e2a) [inline apps/dashboard/urls.py]: As discussed, use a dispatcher also for the edit view.

**Comment by 2e2a** [comment]
lgtm

**Pull request merged**